experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,541 @@
+	}
+}
+
+const (
+	// OffStatus indicates that the input sensor isn't being activated
+	OffStatus TouchStatus = iota
+	// PressedStatus indicates that the input sensor is currently pressed
+	PressedStatus
+	// HeldStatus indicates that the input sensor was pressed and is still held pressed
+	HeldStatus
+	// ReleasedStatus indicates that the input sensor was pressed and is being released
+	ReleasedStatus
+)
+
+const (
+	nbrOfLEDs         = 8
+	strOffStatus      = "Off"
+	strPressedStatus  = "Pressed"
+	strHeldStatus     = "Held"
+	strReleasedStatus = "Released"
+)
+
+const (
+	// reg_LEDLinking - The Sensor Input LED Linking register controls whether a
+	// capacitive touch sensor input is linked to an LED output. If the
+	// corresponding bit is set, then the appropriate LED output will change
+	// states defined by the LED Behavior controls in response to the capacitive
+	// touch sensor input.
+	reg_LEDLinking = 0x72
+	// reg_LEDOutputControl - The LED Output Control Register controls the output
+	// state of the LED pins that are not linked to sensor inputs.
+	reg_LEDOutputControl = 0x74
+)
+
+// Dev is a handle to a cap1188.
+type Dev struct {
+	Opts
+	d             conn.Conn
+	regWrapper    mmr.Dev8
+	isSPI         bool
+	inputStatuses []TouchStatus
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("cap1188{%s}", d.regWrapper.Conn)
+}
+
+// Halt is a noop for the cap1188.
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// InputStatus reads and returns the status of the 8 inputs as an array where
+// each entry indicates a touch event or not.
+func (d *Dev) InputStatus() ([]TouchStatus, error) {
+	// read inputs
+	status, err := d.regWrapper.ReadUint8(0x3)
+	if err != nil {
+		return d.inputStatuses, wrap(fmt.Errorf("failed to read the input values - %s", err))
+	}
+
+	// read deltas (in two's complement, capped at -128 to 127)
+	deltasB := [nbrOfLEDs]byte{}
+	if err = d.regWrapper.ReadStruct(0x10, &deltasB); err != nil {
+		return d.inputStatuses, wrap(fmt.Errorf("failed to read the delta values - %s", err))
+	}
+	deltas := [nbrOfLEDs]int{}
+	for i, b := range deltasB {
+		deltas[i] = int(int8(b))
+	}
+
+	// read threshold
+	thresholds := [nbrOfLEDs]byte{}
+	if err = d.regWrapper.ReadStruct(0x30, &thresholds); err != nil {
+		return d.inputStatuses, wrap(fmt.Errorf("failed to read the threshold values - %s", err))
+	}
+
+	// convert the data into a sensor state
+	var touched bool
+	for i := uint8(0); i < uint8(len(d.inputStatuses)); i++ {
+		touched = isBitSet(status, 7-i)
+
+		// TODO(mattetti): check if the event is passed the threshold:
+		// deltas[i] > int(thresholds[i])
+
+		if touched {
+			if d.inputStatuses[i] == PressedStatus {
+				if d.RetriggerOnHold {
+					d.inputStatuses[i] = HeldStatus
+				}
+				continue
+			}
+			d.inputStatuses[i] = PressedStatus
+		} else {
+			d.inputStatuses[i] = OffStatus
+		}
+	}
+
+	return d.inputStatuses, nil
+}
+
+// LinkLEDs link the behavior of the LEDs to the touch sensors.
```

Comment:
I'd prefer a single function that accepts a bool instead of LInkLEDs/UnlinkLEDs.